<!-- A combination of turtlebot_gazebo and turtlebot_bringup launch files for the Kobuki base-->

<launch>
    <arg name="model_name"/>
    <arg name="x_pos" default="0"/>
    <arg name="y_pos" default="0"/>
    <arg name="yaw" default="0"/>

    <arg name="urdf_file" default="$(find xacro)/xacro.py '$(find cwru_turtlebot)/urdf/kobuki_hexagons_kinect.urdf.xacro' model_name:=$(arg model_name)" />
    <param name="robot_description" command="$(arg urdf_file)" />

    <!-- Gazebo model spawner -->
    <node name="spawn_turtlebot_model" pkg="gazebo_ros" type="spawn_model"
        args="$(optenv robot_initial_pose)
            -unpause
            -urdf 
            -param robot_description
            -model $(arg model_name)
            -x $(arg x_pos)
            -y $(arg y_pos)
            -Y $(arg yaw)"/>

        
    <!-- Velocity muxer -->
    <node pkg="nodelet" type="nodelet" name="mobile_base_nodelet_manager" args="manager"  required="true"/>
    <node pkg="nodelet" type="nodelet" name="cmd_vel_mux"
        args="load yocs_cmd_vel_mux/CmdVelMuxNodelet mobile_base_nodelet_manager"  required="true">
        <param name="yaml_cfg_file" value="$(find turtlebot_bringup)/param/mux.yaml" />
        <remap from="cmd_vel_mux/output" to="mobile_base/commands/velocity"/>
    </node>

    <!-- Mobile Base nodelet-->
    <!-- Only need to enable this if not using Gazebo-->
    <!--node pkg="nodelet" type="nodelet" name="mobile_base" args="load kobuki_node/KobukiNodelet mobile_base_nodelet_manager"-->
        <!--rosparam file="$(find kobuki_node)/param/base.yaml" command="load"/-->
        <!--param name="publish_tf" value="false"/--> <!-- Because we are using the robot_localization package-->
        <!--param name="use_imu_heading" value="false"/--> <!-- Because we are using the robot_localization package-->
    <!--/node-->

    <!--
    <param name="robot/name" value="$(optenv ROBOT turtlebot)"/>
    <param name="robot/type" value="turtlebot"/-->

    <node pkg="diagnostic_aggregator" type="aggregator_node" name="diagnostic_aggregator"  required="true">
        <rosparam command="load" file="$(find turtlebot_bringup)/param/kobuki/diagnostics.yaml"/>
    </node>

  <!-- Bumper/cliff to pointcloud (not working, as it needs sensors/core messages) -->
  <include file="$(find cwru_turtlebot)/launch/kobuki/bumper2pc.launch.xml"/>
</launch>
